testing/pr-testing-guidelines-and-checklists.md

```diff
@@ -0,0 +1,149 @@
+# PR testing guidelines and checklists 
+
+## Introduction
+
+This proposal should cover the PR testing of the repositories concerning the new Aerogear/Feedhenry developments.
+It was based on a discussion between Adam Saleh, Vitali Chepeliuk, Pavel Sturc and Vojtech Sazel.
+
+## Problem Description
+
+We currently don't have a proper set of guidelines for our PR checks and previously we used them mostly just to produce the build-artifacts
+in some fashion. The quality checks have often been added in ad-hoc manner and then disregarded/forgotten about.
+
+## Expectations
+
+If we create a set of coherent guidelines and model our automated PR checks by them we expect
+- to catch regressions early
+- to maintain a good quality of incoming code
+
+## Terms
+
+* relevant repository - repository that could influence ship-able artifacts by it's contents in a bad way. If we really don't want to follow these guidelines, just claim that the repository is currently irrelevant for our project.
+  
+## Proposed Solution
+
+These proposals only concern relevant repositories.
+
+### Repository should have a mandatory review by other person
+
+This is easily enabled in GitHub. It mandates that at least one person gives a PR a green check-mark.
```

Comment:
As mentioned elsewhere I believe it was agreed that we would have 2 LGTM before merge, this stops rushed PRs and ensures knowledge of the change is spread across the team

Reply:
Sounds good. Unfortunately I didn't see a way to enforce two-people review on Github's protected branches. Do you think we should enforce it? There is a possibility to create another check that would use i.e https://developer.github.com/v3/pulls/reviews/ to check the reviewers, but I am not sure it is worth it.

Maybe I would drop the "enable in github" and just have it mentioned in the contributor guidelines and the PR template?